sdk/eidolon_ai_sdk/builtins/logic_units/openapi_helper.py

```diff
@@ -0,0 +1,154 @@
+                  call_fn: callable):
+    actions = []
+    for op_path, op in schema["paths"].items():
+        for operation in operations_to_expose:
+            if operation.matches(op_path):
+                for method_name, method in op.items():
+                    if method_name == operation.method:
+                        name = operation.name
+                        required = []
+                        params = {}
+                        methodParams = None
+                        if "parameters" in method:
+                            methodParams = method["parameters"]
+                            for param in method["parameters"]:
+                                schema_ = param["schema"]
+                                if param["in"] == "query":
+                                    params[param["name"]] = schema_
+                                    if "type" not in schema_:
+                                        schema_["type"] = "string"
+                                    if param["required"]:
+                                        required.append(param["name"])
+                                elif param["in"] == "path":
+                                    params[param["name"]] = schema_
+                                    if "type" not in schema_:
+                                        schema_["type"] = "string"
+                                    required.append(param["name"])
+                                elif param["in"] == "header":
+                                    params[param["name"]] = schema_
+                                    if "type" not in schema_:
+                                        schema_["type"] = "string"
+                                    required.append(param["name"])
+                                else:
+                                    logger.error(f"Unsupported parameter location {param['in']}")
+
+                        if "requestBody" in method:
+                            params["__body__"] = _body_model(schema, method, name)
+                            required.append("__body__")
+                        description = operation.description or _description(method, name)
+                        actions.append(Action(
+                            title=title,
+                            sub_title=method.get("summary", name),
+                            name=name,
+                            action_schema=dict(type="object", properties=params, required=required),
+                            description=description,
+                            tool_call=_call_endpoint(operation.path, operation.result_filters, method_name, methodParams, call_fn)
+                        ))
+    return actions
+
+
+def _description(endpoint_schema, name):
+    description = endpoint_schema.get("description", "")
+    if not description:
+        logger.warning(f"Agent program at {name} does not have a description. LLM may not use it properly")
```

Comment:
this should point to the endpoint rather than an agent program

Reply:
It does.  Warning was wrong.  I fixed that